src/test/java/org/cloudfoundry/promregator/auth/OAuth2XSUAAEnricherTest.java

```diff
@@ -67,31 +68,77 @@ void testAppropriateJWTCall() {
 	@Test
 	void testJWTCallIsBuffered() {
 		this.ams.getOauthTokenHandler().setResponse(this.oAuthServerResponse);
-		
-		OAuth2XSUAAAuthenticationConfiguration authenticatorConfig = new OAuth2XSUAAAuthenticationConfiguration();
-		authenticatorConfig.setClient_id("client_id");
-		authenticatorConfig.setClient_secret("client_secret");
-		authenticatorConfig.setTokenServiceURL("http://localhost:9001/oauth/token");
-		
-		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(authenticatorConfig);
-		
+
+		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(getConfig());
+
 		HttpGet mockGet = Mockito.mock(HttpGet.class);
 		Mockito.when(mockGet.getURI()).thenAnswer(new Answer<URI>() {
 
 			@Override
 			public URI answer(InvocationOnMock invocation) throws Throwable {
 				return new URI("http://localhost/target");
 			}
-			
+
 		});
-		
+
 		// first call will trigger OAuth request
 		subject.enrichWithAuthentication(mockGet);
-		
+
 		// second one should not
 		subject.enrichWithAuthentication(mockGet);
-		
+
 		assertThat(this.ams.getOauthTokenHandler().getCounterCalled()).isEqualTo(1);
 	}
-	
+
+	@Test
+	void testAuthorizationHeaderAddedWhenTokenRetrieved() throws Exception {
+		ClientCredentialsTokenFlow tokenClientMock = Mockito.mock(ClientCredentialsTokenFlow.class);
+		Mockito.when(tokenClientMock.execute())
+				.thenReturn(new OAuth2TokenResponse("someAccessToken", 42l, "somrRefreshToken"));
+
+		ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(getConfig("aScope anotherScope"), tokenClientMock);
```

Comment:
on separator of scopes, please see above (`space` could be wrong here)

Reply:
Adjusted, see above.